src/tests/Helpers/PlanHelperTest.php

```diff
@@ -0,0 +1,335 @@
+			ConstantsHelper::GATEWAY_ID_IN_PAGE_PAY_NOW,
+			ConstantsHelper::DEFAULT_FEE_PLAN
+		) ;
+	}
+
+	public function test_render_checkout_fields_no_plan_eligible() {
+		$this->assertNull($this->alma_plan_helper->render_checkout_fields(
+			array(),
+			array(),
+			ConstantsHelper::GATEWAY_ID_PAY_NOW
+		));
+	}
+
+	public function test_render_checkout_fields_no_plan_eligible_no_gateway_in_plans() {
+
+		$this->assertNull($this->alma_plan_helper->render_checkout_fields(
+			array(),
+			array( ConstantsHelper::GATEWAY_ID_PAY_LATER=> array() ),
+			ConstantsHelper::GATEWAY_ID_PAY_NOW
+		));
+	}
+
+	/**
+	 * @dataProvider checkout_fields_provider
+	 */
+	public function test_render_checkout_fields_in_page($gateway_id, $shoud_received, $should_not_received) {
+
+		$this->gateway_helper_mock->shouldReceive( 'get_alma_gateway_title' )->andReturn( 'test' );
+		$this->gateway_helper_mock->shouldReceive( 'get_alma_gateway_description' )->andReturn( 'test' );
+
+		$plan_helper = \Mockery::spy( PlanHelper::class, [
+			$this->alma_settings_mock,
+			$this->gateway_helper_mock,
+			$this->template_loader_helper_mock,
+			$this->price_factory_mock
+		] )->makePartial();
+		$plan_helper->shouldReceive( $shoud_received )->andReturn( null );
+
+		$this->assertNull( $plan_helper->render_checkout_fields(
+			array(),
+			array( $gateway_id => array( 'test' ) ),
+			$gateway_id
+		) );
+		$plan_helper->shouldHaveReceived( $shoud_received );
+		$plan_helper->shouldNotHaveReceived( $should_not_received );
+	}
+
+	public function checkout_fields_provider() {
+
+		return array(
+			'gateway_id_in_page' => array(
+				ConstantsHelper::GATEWAY_ID_IN_PAGE,
+				'render_fields_in_page',
+				'render_fields_classic'
+			),
+			'gateway_id_in_page_pay_now' => array(
+				ConstantsHelper::GATEWAY_ID_IN_PAGE_PAY_NOW,
+				'render_fields_in_page',
+				'render_fields_classic'
+			),
+			'gateway_id_in_page_pay_later' => array(
+				ConstantsHelper::GATEWAY_ID_IN_PAGE_PAY_LATER,
+				'render_fields_in_page',
+				'render_fields_classic'
+			)
+		);
+	}
+	public function test_render_checkout_fields() {
+
+		$this->gateway_helper_mock->shouldReceive('get_alma_gateway_title')->andReturn('test');
+		$this->gateway_helper_mock->shouldReceive('get_alma_gateway_description')->andReturn('test');
+
+		$plan_helper = \Mockery::spy(PlanHelper::class, [
+			$this->alma_settings_mock,
+			$this->gateway_helper_mock,
+			$this->template_loader_helper_mock,
+			$this->price_factory_mock
+		])->makePartial();
+
+		$plan_helper->shouldReceive( 'render_fields_classic' )->andReturn(null);
+
+		$this->assertNull($plan_helper->render_checkout_fields(
+			array(),
+			array( ConstantsHelper::GATEWAY_ID => array('test') ),
+			ConstantsHelper::GATEWAY_ID
+		));
+
+		$plan_helper->shouldHaveReceived('render_fields_classic');
+		$plan_helper->shouldNotHaveReceived('render_fields_in_page');
+
+	}
+
+
+	public function tear_down() {
```

Comment:
You need to add Mockery::close(); in the tearDown to work fine

Reply:
Done